Sources/StreamChatUI/CommonViews/Attachments/AttachmentsPreviewVC.swift

```diff
@@ -0,0 +1,137 @@
+        }
+    }
+
+    /// The closure handler when an attachment has been removed.
+    open var didTapRemoveItemButton: ((Int) -> Void)?
+    
+    open var selectedAttachmentType: AttachmentType?
+    
+    public private(set) var scrollViewHeightConstraint: NSLayoutConstraint?
+    open private(set) var horizontalConstraints: [NSLayoutConstraint] = []
+    open private(set) var verticalConstraints: [NSLayoutConstraint] = []
+    
+    open private(set) lazy var scrollView: UIScrollView = UIScrollView()
+        .withoutAutoresizingMaskConstraints
+
+    open private(set) lazy var horizontalStackView: ContainerStackView = ContainerStackView(axis: .horizontal, spacing: 8)
+        .withoutAutoresizingMaskConstraints
+
+    open private(set) lazy var verticalStackView: ContainerStackView = ContainerStackView(axis: .vertical, spacing: 8)
+        .withoutAutoresizingMaskConstraints
+    
+    override open func setUpAppearance() {
+        super.setUpAppearance()
+        
+        horizontalStackView.backgroundColor = .clear
+        horizontalStackView.isLayoutMarginsRelativeArrangement = true
+
+        verticalStackView.backgroundColor = .clear
+        verticalStackView.isLayoutMarginsRelativeArrangement = true
+        
+        scrollView.backgroundColor = .clear
+        scrollView.showsHorizontalScrollIndicator = false
+        scrollView.showsVerticalScrollIndicator = false
+    }
+    
+    override open func setUpLayout() {
+        super.setUpLayout()
+        
+        view.embed(scrollView)
+        scrollView.embed(horizontalStackView)
+        scrollView.embed(verticalStackView)
+        
+        horizontalConstraints.append(horizontalStackView.heightAnchor.pin(equalTo: scrollView.heightAnchor))
+        verticalConstraints.append(verticalStackView.widthAnchor.pin(equalTo: scrollView.widthAnchor))
+        
+        scrollViewHeightConstraint = scrollView.heightAnchor.pin(equalToConstant: 0)
+        scrollViewHeightConstraint?.isActive = true
+    }
+    
+    open var attachmentViews: [UIView] {
+        content.enumerated().map { index, attachment in
+            let view = attachment.previewView(components: components).withoutAutoresizingMaskConstraints
+            let cell = components.messageComposer.attachmentCell.init().withoutAutoresizingMaskConstraints
+            cell.embed(attachmentView: view)
+            cell.discardButtonHandler = { [weak self] in self?.didTapRemoveItemButton?(index) }
+            return cell
+        }
+    }
+    
+    open var stackViewAxis: NSLayoutConstraint.Axis {
+        content.first.flatMap { type(of: $0).preferredAxis } ?? .horizontal
+    }
+    
+    override open func updateContent() {
+        super.updateContent()
+        
+        switch stackViewAxis {
+        case .horizontal:
+            setupHorizontalStackView()
+            
+        case .vertical:
+            setupVerticalStackView()
+            
+        @unknown default:
+            break
+        }
```

Comment:
We probably could freeze this enum 🤔

Reply:
@nuno-vieira we don't own it 😛

Reply:
Ahh I see, this is the axis from Apple 😄